packages/forms/src/widgets/MultiSelectTagWidget/MultiSelectTagWidget.js

```diff
@@ -218,132 +163,160 @@ class MultiSelectTagWidget extends React.Component {
 		);
 	}
 
-	scrollDropDownIfRequired(selectedIndex, direction) {
-		const dropDown = this.component.querySelector('.dropdown-menu');
-		const scrollPosition = dropDown.scrollTop;
-		const dropDownHeight = dropDown.clientHeight;
-		const itemStart = (selectedIndex * DROP_DOWN_ITEM_HEIGHT) + DROP_DOWN_PADDING;
-		const itemEnd = ((selectedIndex + 1) * DROP_DOWN_ITEM_HEIGHT) + DROP_DOWN_PADDING;
+	getDropdownItems(suggestions) {
+		if (this.withCategory) {
+			const items = [];
+			const itemsMap = {};
+			const groupBy = this.props.options.groupBy;
+
+			suggestions.forEach((item) => {
+				itemsMap[item[groupBy]] = itemsMap[item[groupBy]] || [];
+				itemsMap[item[groupBy]].push({ ...item, title: item.label });
+			});
+
+			Object.keys(itemsMap).forEach((category) => {
+				items.push({ title: category, suggestions: itemsMap[category] });
+			});
+			return items;
+		} else if (suggestions && suggestions.length > 0) {
+			return suggestions.map(item => ({ ...item, title: item.label }));
+		}
+		return suggestions;
+	}
+
+	updateSuggestions(value, props) {
+		const filterText = value === undefined ? this.state.filterText : value;
+		const currentProps = props || this.props;
+
+		let suggestions = filterOptions(currentProps);
+
+		if (filterText) {
+			const escapedValue = escapeRegexCharacters(filterText.trim());
+			const regex = new RegExp(escapedValue, 'i');
+			suggestions = suggestions.filter(item => regex.test(item.label));
+			if (suggestions.length === 0 && currentProps.schema.createIfNoneMatch) {
+				suggestions.push({ label: `${filterText} (new)`, value: filterText });
+			}
+		}
 
+		this.setState({
```

Comment:
this setState statement still rely on current state of the component to build the new state (https://github.com/Talend/ui/pull/790/files#diff-786130f2dd9d2d717b50077f9be35d44R188), wich is unreliable way unless you use setState with a updated function in wich you will retrieve current component state safely. Please check 
https://reactjs.org/docs/react-component.html#setstate in wich this procedure is better explained.